### PR TITLE
Added test for getting work item types

### PR DESCRIPTION
**Pull request by ldimaggi**

### CODE REVIEW

**Comment by joshuawilson** [comment]
[test]

File: src/app/work-item/work-item.service.ts [hunk 1]
```diff
@@ -49,8 +49,10 @@ export class WorkItemService {
       return this.http
       .get(this.workItemTypeUrl)
       .toPromise()
-      .then((response) => {
-        this.workItemTypes = process.env.ENV != 'inmemory' ? response.json() as WorkItemType[] : response.json().data as WorkItemType[];
+      // .then((response) => {
+       // this.workItemTypes = process.env.ENV != 'inmemory' ? response.json() as WorkItemType[] : response.json().data as WorkItemType[];
+         .then(response => {
+        this.workItemTypes = response.json().data as WorkItemType[];
```
Comment (joshuawilson) [inline src/app/work-item/work-item.service.ts]: @ldimaggi did you change this because you got the tests work without the inmemory?

**Comment by joshuawilson** [comment]
@michaelkleinhenz @sanbornsen please review

File: src/app/work-item/work-item.service.spec.ts [hunk 2]
```diff
@@ -95,6 +96,24 @@ describe('Work Item Service - ', () => {
       });
   }));
 
+
+  it('Get work item types', async(() => {
+    mockService.connections.subscribe((connection: any) => {
+      connection.mockRespond(new Response(
+        new ResponseOptions({
+          body: JSON.stringify(response),
```
Comment (sanbornsen) [inline src/app/work-item/work-item.service.spec.ts]: The response defined up there is for work items, not work item types. You need to define a mock response for the types. The test will pass because it's being tested against the same data but this is not how it should be done.

File: src/app/work-item/work-item.service.spec.ts [hunk 3]
```diff
@@ -95,6 +96,24 @@ describe('Work Item Service - ', () => {
...
+          status: 200
+        })
+      ));
+    });
+
+    apiService.getWorkItemTypes()
+      .then(data => {
+        expect(data).toEqual(resp);
```
Comment (sanbornsen) [inline src/app/work-item/work-item.service.spec.ts]: Same with the resp.

File: src/app/work-item/work-item.service.ts (hunk 1 above)
Comment (sanbornsen) [inline src/app/work-item/work-item.service.ts]: I doubt this will work without in-memory. Our **/workitemtypes** API is still not JSON API formatted.

Comment (sanbornsen) [inline src/app/work-item/work-item.service.ts]: And it doesn't :P 
screen shot 2016-12-03 at 9 18 45 am

**Comment by joshuawilson** [comment]
@sanbornsen I thought that might be the case. Thanks for verifying it.
@ldimaggi please update the PR when you have time.

**Comment by joshuawilson** [comment]
[test]

File: src/app/work-item/work-item.service.ts [hunk 4]
```diff
@@ -40,21 +41,17 @@ export class WorkItemService {
...
-      return new Promise((resolve, reject) => {
-        resolve(this.workItemTypes);
-      });
-    } else {
-      return this.http
-      .get(this.workItemTypeUrl)
+ getWorkItemTypes(): Promise<WorkItemType[]> {
+    this.logger.log('WorkItemService using url ' + this.workItemTypeUrl);
+    return this.http
+      .get(this.workItemTypeUrl + '.2', {headers: this.headers})
```
Comment (joshuawilson) [inline src/app/work-item/work-item.service.ts]: I think the point of having a variable for the URL is so that you don't have to change it in all the places it is used. Adding '.2' is the type of change this is meant to avoid.

File: src/app/work-item/work-item.service.ts [hunk 5]
```diff
@@ -40,21 +41,17 @@ export class WorkItemService {
...
       .toPromise()
-      .then((response) => {
-        this.workItemTypes = process.env.ENV != 'inmemory' ? response.json() as WorkItemType[] : response.json().data as WorkItemType[];
+      //.then(response => process.env.ENV != 'inmemory' ? response.json() as WorkItem[] : response.json().data as WorkItem[])
+      .then(response => {
+        this.workItemTypes = response.json().data as WorkItemType[];
```
Comment (joshuawilson) [inline src/app/work-item/work-item.service.ts]: Why did you change it to not use or allow the use of "inmemory"?

**Comment by ldimaggi** [comment]
Closing this PR - tests will be added in a diff PR - too many conflicts between this PR and others. Thx!
